effekt/shared/src/main/scala/effekt/core/StaticArguments.scala

```diff
@@ -96,7 +97,7 @@ object StaticArguments {
         dropStatic(staticB, blockLit.cparams),
         dropStatic(staticV, blockLit.vparams),
         dropStatic(staticB, blockLit.bparams),
-        rewrite(blockLit.body)
+        rewrite(blockLit.body)(using enterFunction(id))
```

Comment:
The recursive calls were never rewritten and thus all programs had the form:

```
def wrapper() = {
  def worker() = {
    ... wrapper() ...
  }
  worker()
}
```
which resulted in a useless `worker`.